tests/integration/data_sources_and_expectations/data_sources/test_snowflake.py

```diff
@@ -0,0 +1,270 @@
+from datetime import datetime, timezone
+
+import pandas as pd
+import pytest
+
+from great_expectations.compatibility.snowflake import SNOWFLAKE_TYPES
+from great_expectations.compatibility.sqlalchemy import sqltypes
+from great_expectations.expectations import (
+    ExpectColumnDistinctValuesToContainSet,
+    ExpectColumnSumToBeBetween,
+    ExpectColumnValuesToBeBetween,
+    ExpectColumnValuesToBeOfType,
+)
+from tests.integration.test_utils.data_source_config import SnowflakeDatasourceTestConfig
+from tests.integration.test_utils.data_source_config.snowflake import SnowflakeBatchTestSetup
+
+
+class TestSnowflakeDataTypes:
+    """This set of tests ensures that we can run expectations against every data
+    type supported by Snowflake.
+
+    https://docs.snowflake.com/en/sql-reference/intro-summary-data-types
+    """
+
+    COLUMN = "col_a"
+
+    @pytest.mark.snowflake
+    def test_number(self):
+        column_type = (
+            SNOWFLAKE_TYPES.NUMBER
+        )  # snowflake specific type equivalent to DECIMAL, NUMERIC
+        batch_setup = SnowflakeBatchTestSetup(
+            config=SnowflakeDatasourceTestConfig(column_types={self.COLUMN: column_type}),
+            data=pd.DataFrame({self.COLUMN: [1, 2, 3, 4]}),
+            extra_data={},
+        )
+        with batch_setup.batch_test_context() as batch:
+            result = batch.validate(
+                expect=ExpectColumnSumToBeBetween(
```

Comment:
Probably room to DRY up some of this by pulling out the expectation/dataframe into constants, but I'd rather just get it merged and circle back.